Public/Src/Engine/UnitTests/Processes.Detours/PipExecutorDetoursTest.cs

```diff
@@ -6163,6 +6163,46 @@ private static FileArtifact WriteFile(PathTable pathTable, AbsolutePath filePath
             return FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, expandedPath));
         }
 
+        [Fact]
+        public async Task CallDeleteWithoutSharing()
+        {
+            var context = BuildXLContext.CreateInstanceForTesting();
+            var pathTable = context.PathTable;
+
+            using (var tempFiles = new TempFileStorage(canGetFileNames: true, rootPath: TemporaryDirectory))
+            {
+                var untrackedFile = tempFiles.GetFileName(pathTable, "untracked.txt");
+                WriteFile(pathTable, untrackedFile, "real");
+
+                var process = CreateDetourProcess(
+                    context,
+                    pathTable,
+                    tempFiles,
+                    argumentStr: "CallDeleteWithoutSharing",
+                    inputFiles: ReadOnlyArray<FileArtifact>.Empty,
+                    inputDirectories: ReadOnlyArray<DirectoryArtifact>.Empty,
+                    outputFiles: ReadOnlyArray<FileArtifactWithAttributes>.Empty,
+                    outputDirectories: ReadOnlyArray<DirectoryArtifact>.Empty,
+                    untrackedScopes: ReadOnlyArray<AbsolutePath>.FromWithoutCopy(untrackedFile.GetParent(pathTable)));
+
+                string errorString = null;
+                SandboxedProcessPipExecutionResult result = await RunProcessAsync(
+                    pathTable: pathTable,
+                    ignoreSetFileInformationByHandle: false,
+                    ignoreZwRenameFileInformation: false,
+                    monitorNtCreate: true,
+                    ignoreRepPoints: false,
+                    disableDetours: false,
+                    context: context,
+                    pip: process,
+                    errorString: out errorString);
+
+                VerifyNoFileAccesses(result);
+                VerifySharingViolation(context, result);

```

Comment:
>VerifySharingViolation(context, result); [](start = 16, length = 40)

Do we have test coverage for the other side of this? That when you are interacting with files that are tracked fileshare delete is added so other processes can delete?

Reply:
I remove `FILE_SHARE_DELETE` altogether, so that we don't have any modification except what is explicitly requested, e.g., downgrading and force read sharing